APE21.rst

```diff
@@ -0,0 +1,187 @@
+
+* No longer designate any releases as LTS.
+* Keep the 6-month release cycle and bump releases to x.0 every 2 years (with
+  the possibility of allowing this to be more or less than 2 years if there is a
+  consensus amongst core package maintainers that this can be changed)
+* Refer to the x.0 releases every 2 years (e.g., v6.0) as "major"
+  releases, other releases every 6 months (e.g., v6.1) as "minor", and bug fixing
+  releases (e.g., v6.1.2) as "bugfix" releases.
+* Minimize as much as possible any changes that break API or change results in
+  minor releases, but be pragmatic and occasionally allow such changes if they would
+  not affect many users.
+* Add deprecation warnings or future warnings in minor and major releases but only
+  go through with the breaking changes in major releases (either the next major
+  release following when a warning was added, or subsequent ones if the changes
+  are deemed to be high impact enough to wait longer).
+
+With this in place, we would effectively be striving to maximize stability over
+two-year periods but users would see deprecation warnings to prepare them for
+changes. This is in contrast to the current LTS system where users might not see
+a deprecation warning until the next LTS release as deprecations are not typically
+backported.
+
+Guidelines on deprecations and API changes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The main proposed guidelines are as follows:
+
+* API changes should only be carried out in major releases.
+* Any API changes should be preceded by deprecation warnings that should be
+  visible to users in at least one minor release cycle prior to the API change
+  being carried out.
+* New deprecations should not be introduced in bugfix releases - this will
+  ensure that deprecations will therefore be emitted for at least 6 months
+  before a major release.
+* If developers wish to make an API change at a point in time where the next
+  release is a major release, they should introduce the deprecation in the major
+  release and carry out the change in the following major release.
+* The following will not be considered to be part of the API and can therefore
+  be changed in bugfix and minor releases:
+
+    * Changes to warning messages
+    * Changes to exception messages
+    * Changes to the ``__repr__`` or ``__repr_*__`` of objects
+
+  However, exception and warning *types* will be considered to be part of the
+  API.
+
+As with all guidelines, there are exceptions where we can deviate from these guidelines:
+
+* API changes can be carried out in major releases without deprecation if it is
+  not possible to have a deprecation phase due to the nature of the change.
+* Changes breaking or changing the behavior of code may be needed in order to
+  fix bugs. Such changes can be made in bugfix or minor releases and do not need
+  to be considered API changes.
+* Changes to exception or warning types may occasionally be done in minor
```

Comment:
Maybe generalize slightly, to make it more consistent with what you had on top: "Small-impact changes such as modifications in exception or warning types..."

Reply:
At this point (10 years in) I would argue that changing exception or warning type base classes should be very rare. What might be more common and even encouraged would be moving to more specific exceptions/warnings that are derived from the original base class. I.e. `class IoAsciiSomeSpecificError(ValueError)`. This seems like a non-breaking change that should be encouraged, but changing `ValueError` to `TypeError` should be strongly discouraged.

Reply:
Agreed to rephrase, e.g., "Changes to more specific exception or ..."

Reply:
@nstarman - I think what @taldcroft and I wrote is consistent with your wish! Please suggest rephrasing if it doesn't feel like it is!

Reply:
Yes, I think we're saying the same thing. An exception that is now `ValueError` would become a custom exception that inherits from `ValueError` so that existing `except ValueError` statements would typically work.